spec/integration/schema/macros/hash_spec.rb

```diff
@@ -98,6 +98,28 @@
       end
     end
 
+    context 'with custom coercible type' do
+ 
+      subject(:schema) do
+        ExpirationDate = Types::DateTime.constructor { |value| value.to_time.round.to_datetime }
+        Dry::Schema.Params do
+          required(:unnested_dated).value(ExpirationDate)
+          required(:foo).hash do
+            required(:bar).hash do
+              required(:nested_date).value(ExpirationDate)
+            end
+          end
+        end
+      end
+
+      let(:input) { {foo: {nested_date: '2021-11-11T00:00:00+00:00'}, unnested_date: '2021-11-11T00:00:00+00:00'  } }
```

Comment:
it should be `foo: { bar:` according to the schema

Reply:
Thanks!